utils/utils.go

```diff
@@ -0,0 +1,356 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Each resource request will be fetched from the Kubernetes API Server at a
+// max of 12 retries, with 10 seconds in between each attempt.
+// This creates a max wait time of upto 2 minutes that a resource request
+// must successfully return within, before moving on.
+
+// MaxRetries is the maximum number of retries that a resource will be
+// attempted to be fetched from the Kubernetes API Server. All transient
+// request attempts count towards the retries.
+const MaxRetries = 12
+
+// RetryInterval is the number of seconds to sleep in between requests
+// to the Kubernetes API Server.
+const RetryInterval = 10
+
+// EKSSmokeTest runs a checklist of operational successes required to deem the
+// EKS cluster as successfully running and ready for use.
+func EKSSmokeTest(t *testing.T, clientset *kubernetes.Clientset, desiredNodeCount int) {
+	// Print API Server info.
+	APIServerVersionInfo(t, clientset)
+
+	// Run all tests.
+	AssertEKSConfigMapReady(t, clientset)
+	AssertAllNodesReady(t, clientset, desiredNodeCount)
+	AssertAllPodsReady(t, clientset)
+}
+
+// APIServerVersionInfo prints out the API Server versions.
+func APIServerVersionInfo(t *testing.T, clientset *kubernetes.Clientset) {
+	version, err := clientset.DiscoveryClient.ServerVersion()
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("API Server Version: %s.%s\n", version.Major, version.Minor)
+	fmt.Printf("API Server GitVersion: %s\n", version.GitVersion)
+}
+
+// AssertEKSConfigMapReady ensures that the EKS aws-auth ConfigMap
+// exists and has data.
+func AssertEKSConfigMapReady(t *testing.T, clientset *kubernetes.Clientset) {
+	awsAuthReady, retries := false, MaxRetries
+	configMapName, namespace := "aws-auth", "kube-system"
+	var awsAuth *v1.ConfigMap
+	var err error
+
+	for !awsAuthReady && retries > 0 {
+		awsAuth, err = clientset.CoreV1().ConfigMaps(namespace).Get(configMapName, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Waiting for '%s' ConfigMap to be returned. Retrying...\n", configMapName)
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+			continue
+		}
+		awsAuthReady = true
+		break
+	}
+
+	require.Equal(t, true, awsAuthReady, "EKS ConfigMap '%s' does not exist in namespace '%s'", configMapName, namespace)
+	require.NotNil(t, awsAuth.Data, "%s ConfigMap should not be nil", configMapName)
+	require.NotEmpty(t, awsAuth.Data, "'%s' ConfigMap should not be empty", configMapName)
+	fmt.Printf("EKS ConfigMap '%s' exists and has data\n", configMapName)
```

Comment:
Why do you need to `Printf` and `Logf` the same message?

Reply:
tl;dr: `t.Logf` in Go [only outputs](https://github.com/golang/go/issues/24929#issuecomment-384484654) success/failure after tests have completed, or failed - not live.

The `Printf's` are a means of showing transient output as the tests are running live, given that we're spinning up many different types of clusters, and sometimes its better to know more detail upfront about what's happening right or wrong, versus waiting 45+ min for builds to finish or crash to get the output.

This is more of a personal "quality of life" change, than anything else.